### ISSUE TITLE
AGMBallistic possible bug

### ISSUE THREAD

**Issue by NextStarIndustries** [PapaJoesSoup/BDArmory#274]
##### ISSUE TYPE
 - Support 	    ( Need help getting something to work)
 - Bug Report   (Something is broken)

##### KSP and BDA Version
KSP 1.3

BDAc 0.3.0.0

##### OS / ENVIRONMENT
Windows 10 fully updated

##### SUMMARY
It seems when using AGMBallistic, missiles that go above 20 km high lose their target and continue off into space at a high rate of acceleration, like the engine isn't cut off, although it clearly has been.

##### STEPS TO REPRODUCE
Launch a longer ranged missile from around 70 to 80 km from the target, lock GPS to target and fire missile. If it goes above 20 km high it happens.

##### Log Files
Log file has nothing about it since it is only affecting the missile.

**Comment by NextStarIndustries** [comment]
I can link a video if it will help.

**Comment by jrodrigv** [comment]
Thanks for opening the issue. 

Currently, there is no missiles capable of flying at high altitudes where the atmosphere density is very low. 

However, we are aware of this and we will be working on a solution for this.

**Comment by NextStarIndustries** [comment]
Should I close this seeing how it's just a limitation and not an issue?

**Comment by gomker** [comment]
We can keep it open to track JR's progress with the new functionality

**Comment by jrodrigv** [comment]
Sounds good. I'm working on this issue at the moment :)

**Comment by jrodrigv** [comment]
Hi all,
I think I have fixed the infinite fuel issue here.
https://github.com/jrodrigv/BDArmory/releases/tag/dev28082017

But I would like someone else to test it.

@NextStarIndustries Would you mind to test it?

**Comment by NextStarIndustries** [comment]
Thanks I'm testing now.

**Comment by NextStarIndustries** [comment]
@jrodrigv First WOW! Everything is working great, really great.
 I did have to do a recompile for my stuff, missile launcher changed, but after that perfect. I do have to do some adjustments to my configs now, but everything I tested worked amazingly.
 I really like the throttling, may need a new "exhaust" made, some angles of the camera caused weird texture effects when it was throttling.
 I have yet to find a limit to how far out I can launch weapons, I'm out to 600 km and still hits perfectly on target with proper config files.
Thank you for the space stuff as well, this will be really fun to play with.
I believe this will now allow true ICBMs as well, which is really exciting, and I plan to do more testing today with that aspect in mind.
I do have a question about the BallisticOverShootFactor how is this to be used properly? What is it representing, a distance?

**Comment by PapaJoesSoup** [comment]
There is a limit on range, based on the accumulation of floating point errors, you will see that manifest in the missile starting to shake itself a part as I understand from our conversations, but with 600km plus, we are really getting out there, eh?

I'm really excited by these new features, and as we get a better understanding older bugs are starting to disappear as well.  I'm sure @jrodrigv will have more to say on the BallisticOverShootFactor.  we are working on eliminating the need for that.

**Comment by NextStarIndustries** [comment]
Thanks for the info @PapaJoesSoup , I'll keep that in mind while testing.
I was going to do an update soon, but after getting my hands on this today, I'll be waiting for the team to release the next version with these new features implemented. :) Time to start modeling some more long range weapons.
@PapaJoesSoup one more question will this be a release for KSP 1.3 or 1.3.1?

**Comment by PapaJoesSoup** [comment]
We have tested on 1.3 and 1.3.1  There was one small issue found (with a localization change) so we will need to recompile for 1.3.1  therefore, I "think" we will be releasing for 1.3.1.  

KSP 1.3.1 was a maintenance update, so no earth shattering things changed (other than some nice quality of life improvements as shown in the release notes).  It is looking pretty stable from our point of view.  There are some other features in the works that are in the midst of finalization and QA testing, so I think we can safely say 1.3.1.

**Comment by jrodrigv** [comment]
@NextStarIndustries Thanks for the feedback!

Regarding the BallisticOverShootFactor. It is a property that affect the distance the missile will fly.
A greater value will make the missile to follow an extended path, thus achieving a more vertical hit against the target.

Moreover, this property is more important for modular ballistic missile because modular missiles are not as accurate in vacuum as standard missiles, so if we want the perfect hit we will need to adjust the value to follow the perfect ballistic trajectory.

**Comment by NextStarIndustries** [comment]
@jrodrigv After distance testing some more I made it out to 850 km and then experienced the shake with my "Honest John" missile just as @PapaJoesSoup said. I thought I may have had the torque to high, but after playing with the number it had no effect. I don't have any weapons that require these kind of distances anyway, but it's nice to have the option to create some.
Engine cut off is spot on, flight path is near perfect, and missile holds GPS targets spot on.
The new cruise missile I have in the works has also tested out perfectly at any altitude I choose. I have it able to be launched from over 500 km as well.
I have tried everything I can think of to cause an issue and so far not one has manifested.
@PapaJoesSoup A release and forget ICBM would be fun, but manually flying them(or using MechJeb) to target is just as fun and more interactive anyway. :)
